sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1052,15 +1049,15 @@ object functions {
    * @group agg_funcs
    * @since 3.5.0
    */
-  def std(e: Column): Column = stddev(e)
+  def std(e: Column): Column = call_function("std", e)
 
   /**
    * Aggregate function: alias for `stddev_samp`.
    *
    * @group agg_funcs
    * @since 1.6.0
    */
-  def stddev(e: Column): Column = withAggregateFunction { StddevSamp(e.expr) }
+  def stddev(e: Column): Column = call_function("stddev", e)
```

Comment:
This one may is a break change. Even if the behavior is more suitable.

Reply:
good catch! it was since 1.6.0.
but what about keeping this change, otherwise it make SQL/Connect(both Scala Client and Python Client) different from vanilla Spark(Python and Scala)

Reply:
according to https://github.com/apache/spark/blob/9d28bef2f70b06cbb2f50a6814f8433fa344052e/sql/core/src/main/scala/org/apache/spark/sql/functions.scala#L54-L55, I think we can change such inconsistence to make `stddev("id")` consistent with `expr("stddev(id)"))`

```
scala> df.select(stddev("id"), expr("stddev(id)"))
res0: org.apache.spark.sql.DataFrame = [stddev_samp(id): double, stddev(id): double]
```

also cc @HyukjinKwon

Reply:
I think we should fix the alias to `stddev` for the cases that are not.

Reply:
thanks, there are still some cases introduced in older versions, I will take a look later.
For version before 3.5.0, this PR only fix `stddev` due to its relationship to the newly added `std`